### ISSUE TITLE
Questions regarding TMP settings

### ISSUE THREAD

**Issue by Bearmancer** [onemen/TabMixPlus#77]
I have been systematically going through all the option in TMP and there are a few that I don't understand even after reading their descriptions.

1) Move tab from 'Switch to tab' next to current one - alt+tab switches tab in the order set in Firefox.

2) Prevent tab from closing:
	Focus content when loading non blank page.
Even after selecting the option, the focus in the new tab is still on the address bar.

3) Controls tabs and windows merging when tabs are closed 

I don't see any option of merging tabs. I opened two windows simultaneously with different tabs but they didn't merge into one window when I re-opened firefox. The second instance was discarded.

4) Both options look exactly the same. What am I missing?

5) Use middle mouse click on Closed Tabs List button to restore all closed tabs at once.

6) Use right mouse click on recently closed tabs list menu to open the context menu seen in the image - Neither of the features work.

7) Ctrl-Tab displays a tab list popup menu - pressing Ctrl-Tab will display a popup menu list of recently closed tabs 
Pressing ctrl+tab after enabling this option does not show recently closed tab.

8) When scrolling, show part of next out-of-sight tab - for single row options
Does this option require enabling tab scrolling through Firefox's about:config?

9) Mouse click (down and release) to select a tab
That just opens the tab in a new window.

10) Show Tab Context Menu on tabbar 
I can't see any difference toggling the two options.

11) I disabled 'Send Tab to Device' but the option still appears on the tab bar context menu.

I am posting my questions here because I can't create an account on the forums. There's no image for captcha or a box to enter text. I disabled uBlock Origin too - no difference.

**Comment by onemen** [comment]
Did you visit our [help pages](http://tabmixplus.org/support/viewtopic.php?t=3)

**Comment by onemen** [comment]
What is your Firefox version?

**Comment by onemen** [comment]
First I would like to thank you for the thorough review 👍, although I have some questions I did manage to find some bugs.

> I can't create an account on the forums

I need to update the forums software on our [server host](https://www.bluehost.com/). 

Regarding your points from above:

> 1) Move tab from 'Switch to tab' next to current one - alt+tab switches tab in the order set in Firefox.

```From our docs: When using 'Switch to tab' from address bar popup, move the tab you switching to next to the currently selected tab.```

> 2) Prevent tab from closing:  
	Focus content when loading non blank page.
Even after selecting the option, the focus in the new tab is still on the address bar.

That is a **bug**, when loading page with content after closing last tab the focus should be on the page, I will working on it now.

> 3) Controls tabs and windows merging when tabs are closed 

This is multi step process:
 * Set your preferences in **Options > Events > Tab Merging**
 * If you like to merge individual tabs, Go to in **Options > Mouse > Mouse Clicking** and set one of the option to: ```Selects the tab for merging```.
 * When you want to merge windows/tab you need to activate the command, You can do it from tab or main context menu or by using shortcut. Look in **Options > Menu** for ```Merge windows```

> 4) Both options look exactly the same. What am I missing?

I don't understand what are the options you refers to in 4.

> regarding 5) 6) 7)

I don't know what the issue, if the commands doesn't work at all for you, open the console from top menu **Tools > Browser Tools > Browser Console** clear it, then try the command and report if you see any messages.
Also Report your Firefox version and operating system.

> 8) When scrolling, show part of next out-of-sight tab - for single row options
Does this option require enabling tab scrolling through Firefox's about:config?

NO. It just help to make sure that the right most visible tab is clickable.

> 9) Mouse click (down and release) to select a tab
That just opens the tab in a new window.

Are you sure, that when you click on a tab Firefox open new window?
Firefox default behavior is to select a tab on mouse down, this options change it so the tab selection take place only after mouse up. It can help if you would like to drag tab around without selecting it.

>10) Show Tab Context Menu on tabbar 
I can't see any difference toggling the two options.

When this option is off Firefox shows the default context menu for this area, when this option is on Firefox shows the **tab context menu**. you can control the content of tab context menu from **Options > Menu > Tab Context Menu**

>11) I disabled 'Send Tab to Device' but the option still appears on the tab bar context menu.

That is a **Bug**, one line was missing from our style sheet.

**Comment by Bearmancer** [comment]
TMP Preferences: https://we.tl/t-eRGab1uBRD

```From our docs: When using 'Switch to tab' from address bar popup, move the tab you switching to next to the currently selected tab.```

Yeah I really have no idea what this does. Pressing alt+tab regardless of the option has address bar on the focus. Is this a bug?

```When loading page with content after closing last tab the focus should be on the page.
What is the address of the tab you loading after closing last tab?```

Any page. Including this very link. Focused on address bar.

Going slowly because otherwise there are too many concurrent threads to address.

TMP Preferences: https://we.tl/t-WeVjGTVV5O

**Comment by Bearmancer** [comment]
My Firefox details: 93.0b6 64 bit

**Comment by onemen** [comment]
> `From our docs: When using 'Switch to tab' from address bar popup, move the tab you switching to next to the currently selected tab.`
> 
> Yeah I really have no idea what this does. Pressing alt+tab regardless of the option has address bar on the focus. Is this a bug?
![switchtotab1](https://user-images.githubusercontent.com/3650909/133887436-4ac0b285-b98b-4c97-9695-62109afe3efc.png)

**Comment by Bearmancer** [comment]
> 
> 
> > `From our docs: When using 'Switch to tab' from address bar popup, move the tab you switching to next to the currently selected tab.`
> > Yeah I really have no idea what this does. Pressing alt+tab regardless of the option has address bar on the focus. Is this a bug?
> > ![switchtotab](https://user-images.githubusercontent.com/3650909/133887469-d36ee49c-f735-45d0-8135-6c6fe64a0711.png)
> 
> After testing it, `move the tab you switching to next to the currently selected tab` does not work

Ok when I press Ctrl+Tab ON switch to tab, the tab switcher vertical list pops up. Not whatever is supposed to happen.

**Comment by onemen** [comment]
The preference ```Move tab from 'Switch to tab' next to current one``` refers to ```Switch to tab``` label that appears in the popup below the address bar when you type an address that matches address of another open tab. (see image above)
The preference has nothing to do with Ctrl-Tab or Alt-Tab actions.

As i mention earlier today, there is a bug in this feature that will be fix in the next pre-releas

**Comment by onemen** [comment]
Just updated the doc regarding [Switch to tab](http://tabmixplus.org/support/viewpage.php?t=3&p=events-new-tabs) and [Ctrl-Tab](http://tabmixplus.org/support/viewpage.php?t=3&p=events-tab-features)

**Comment by Bearmancer** [comment]
> 
> 
> The preference `Move tab from 'Switch to tab' next to current one` refers to `Switch to tab` label that appears in the popup below the address bar when you type an address that matches address of another open tab. (see image above)
> The preference has nothing to do with Ctrl-Tab or Alt-Tab actions.
> 
> As i mention earlier today, there is a bug in this feature that will be fix in the next pre-releas

So it causes the tab to stick next to the currently open tab right? When it works I might understand what it does.

Moving on:
I am confused because there are so many options but which option is #3 specifically? I may have misunderstood what tab merging. Is there an option so that if I had three Firefox windows opens and I close all three windows the next time Firefox opens up all three windows are merged at once?

**Comment by onemen** [comment]
>  Is there an option so that if I had three Firefox windows opens and I close all three windows the next time Firefox opens up all three windows are merged at once

No.

There is an options to merge all window with one shortcut  **Ctrl+Shift+U**

![merge](https://user-images.githubusercontent.com/3650909/133933360-69e9f6eb-9e11-4f30-ad84-b97416c58c44.png)

**Comment by Bearmancer** [comment]
Don't remember #4 either.

#5 is fixed. I had to change what middle mouse does for no. 5.

#7 basically means that the tab switching will happen in order I open them last right? Well that works then.

**Comment by onemen** [comment]
Don't use # character followed by and integer, Github a automatically convert it to a link.
Use 3) instead

**Comment by onemen** [comment]
> 7) basically means that the tab switching will happen in order I open them last right? Well that works then.

Yes, instead of using Firefox build in popup that cover the screen with all tabs thumbnail, we just show tab list in the middle of the screen.
